x-pack/auditbeat/module/system/user/user.go

```diff
@@ -381,22 +403,54 @@ func (ms *MetricSet) reportChanges(report mb.ReporterV2) error {
 	return nil
 }
 
-func userEvent(user *User, eventType string, eventAction string) mb.Event {
+func userEvent(user *User, eventType string, action eventAction) mb.Event {
 	return mb.Event{
 		RootFields: common.MapStr{
 			"event": common.MapStr{
 				"type":   eventType,
-				"action": eventAction,
+				"action": action.String(),
 			},
 			"user": common.MapStr{
 				"id":   user.UID,
 				"name": user.Name,
 			},
+			"message": userMessage(user, action),
 		},
 		MetricSetFields: user.toMapStr(),
 	}
 }
 
+func userMessage(user *User, action eventAction) string {
+	var actionString string
+	switch action {
+	case eventActionExistingUser:
+		actionString = "Existing"
```

Comment:
These could have been a method on the `eventAction` type, like `Render()` or something like that. Just an idea.

Reply:
I agree in general. In this case I prefer this since the value is only ever used right after being filled. A method would separate this flow, and probably to different parts of the file.